src/models/transactions/checkCash.ts

```diff
@@ -0,0 +1,37 @@
+import { ValidationError } from "../../common/errors";
+import { Amount, IssuedCurrencyAmount } from "../common";
+import { BaseTransaction, verifyBaseTransaction } from "./common";
+
+export interface CheckCash extends BaseTransaction {
+    TransactionType: "CheckCash";
+    CheckID: string;
+    Amount?:	Amount;
+    DeliverMin?: Amount;
+}
+
+/**
+ * Verify the form and type of an CheckCash at runtime.
+ * 
+ * @param tx - An CheckCash Transaction
+ * @returns - Void.
+ * @throws - When the CheckCash is Malformed.
+ */
+ export function verifyCheckCash(tx: CheckCash): void {
+    verifyBaseTransaction(tx)
+
+    const isIssuedCurrency = (obj: IssuedCurrencyAmount): boolean => {
+        return Object.keys(obj).length === 3 
+            && typeof obj.value === 'string'
+            && typeof obj.issuer === 'string'
+            && typeof obj.currency === 'string'
+    }
+
+    if (tx.Amount !== undefined && typeof tx.Amount !== 'string' && !isIssuedCurrency(tx.Amount))
+        throw new ValidationError("CheckCash: invalid Amount")
+
+    if (tx.DeliverMin !== undefined && typeof tx.DeliverMin !== 'string' && !isIssuedCurrency(tx.DeliverMin))
+        throw new ValidationError("CheckCash: invalid DeliverMin")
+
```

Comment:
Related to above, I don't think 'string' is a valid type for Amount/DeliverMin (If it is and I'm wrong this is good)

Reply:
(If Amount is the correct type for this, could you add a helper function to common where the Amount type is to validate 'Amount' types so we don't need to know that Amounts are String | IssuedCurrencyAmount in our validation checks?)